### DISCUSSION TITLE
Longer timeout in Express application

### DISCUSSION THREAD

**Discussion by BazzoKelesh** [vercel/community#4232] [Help]
### Summary

How can I increase the timeout when I can't use build and functions property together?
`{
  "builds": [
    {
      "src": "index.js",
      "use": "@vercel/node"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "index.js"
    }
  ],
  "functions": {
    "index.js": {
      "maxDuration": 300
    }
  }
}
`

### Steps to Reproduce

Adding the functions property to vercel.json which already contains build.

**Comment by amyegan** [comment]
Hi @BazzoKelesh! The [builds](https://vercel.com/docs/projects/project-configuration#builds) property is a legacy setting. For most cases, it is recommended that you update to the use the [functions](https://vercel.com/docs/projects/project-configuration#functions) property instead. Keep in mind that max execution duration can only go **up to the maximum limit of your plan**. The [maxDuration](https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration) limit is currently 10 seconds for Hobby and 300 seconds for Pro.



The [routes](https://vercel.com/docs/projects/project-configuration#routes) property is also legacy and only intended for advanced integration purposes, such as the [Build Output API](https://vercel.com/docs/build-output-api/v3). It should be updated to [redirects](https://vercel.com/docs/projects/project-configuration#redirects) or [rewrites](https://vercel.com/docs/projects/project-configuration#rewrites) instead.

You can find more info about the recommended configuration for using Express with Vercel in this guide: https://vercel.com/guides/using-express-with-vercel

**Reply by BazzoKelesh** [reply]
Thanks for the reply @amyegan. Not sure if I get it correctly from what I read. To make it work, I should remove the build and routes property and only have the function property as I had above? And I need to refactor my folder structure for the express api (putting it inside an api folder)? I can't keep the index.js in the root with the new vercel.json settings? Because it doesn't build the app when the index.js is still in the root when I remove the build and routes property, it shows the source code instead.

It's an api that is in early production, with active users. I want to avoid changing the current folder setup. While the only feature I want is to increase the timeout. How would the migrated vercel.json look like with the current folder structure (root index.js) and increased timeout?

Thanks!

**Comment by Thribs** [comment]
See [this answer](https://github.com/orgs/vercel/discussions/6684#discussioncomment-9509661)
